<img src="http://imgur.com/1ZcRyrc.png" style="float: left; margin: 20px; height: 55px">

## Measures of Central Tendency: Independent Practice

---

Complete the questions using the distribution defined with the **`dist`** variable in the next cell.

In [1]:
import numpy as np
from scipy.stats import mode

dist = [33.3, 8.0, 2.7, 7.0, 11.3, 10.2, 38.4, 14.7, 46.9, 6.6, 38.6, 41.1, 20.4, 19.5,
        102.6, 111.5, 21.8, 27.0, 30.3, 9.3, 19.6, 31.1, 48.8, 8.9, 11.4, 0.6, 23.5,
        8.0, 27.5, 69.7, 11.5, 15.5, 1.4, 7.0, 4.6, 6.0, 16.0, 17.2, 6.3, 26.2, 23.9,
        45.9, 55.4, 100.8, 22.9, 75.5, 115.5, 58.9, 10.5, 92.6, 12.4, 102.7, 10.2, 167.6,
        143.6, 132.3, 134.2, 39.4, 49.7, 12.0, 27.0, 1.3, 4.7, 5.0, 24.1, 21.5, 33.1,
        45.6, 46.3, 15.8, 30.7, 15.7, 6.8, 64.6, 39.1, 2.9, 8.0, 5.4, 33.4, 40.0, 22.7,
        24.2, 25.7, 85.0, 26.2, 3.3, 20.7, 117.4, 66.8, 44.3, 37.9, 71.8, 161.4, 45.9,
        4.6, 6.6, 12.6, 24.4, 21.4, 33.6, 33.7, 18.8, 7.6, 36.3, 1.8, 83.2, 75.1, 71.3,
        32.9, 30.6, 16.0, 6.9, 11.0, 9.4, 11.7, 13.8, 39.4, 4.3, 38.1, 60.1, 78.2, 49.6,
        14.3, 8.0, 19.3, 30.8, 32.5, 29.6, 28.3, 32.4, 35.7, 36.7, 10.0, 8.5, 12.2, 22.6,
        32.1, 25.7, 14.1, 20.9, 23.8, 29.5, 2.9, 42.3, 19.9, 47.5, 15.9, 6.0, 19.3, 38.2,
        23.5, 2.7, 27.0, 49.9, 43.5, 31.7, 39.1, 15.1, 18.0, 33.8, 32.1, 50.7, 28.8, 11.6,
        74.8, 16.2, 8.3, 36.3, 44.0, 15.6, 48.9, 7.3, 8.2, 29.8, 38.7, 3.9, 7.2, 5.1, 23.7,
        48.4, 92.0, 64.3, 97.4, 99.1, 18.0, 7.1, 14.7, 47.1, 41.3, 44.2, 6.6, 44.4, 9.7,
        15.5, 23.7, 34.5, 19.9, 58.6, 35.0, 66.1, 74.5, 45.2]

##### Q1: Calculate the mean of `dist`.

In [2]:
mean = sum(dist) / len(dist)
print(mean)

33.99554455445545


or

In [3]:
mean = np.mean(dist)
print(mean)

33.99554455445545


##### Q2: Calculate the median of `dist`.

In [4]:
median = np.median(dist)
print(median)

25.95


##### Q3: Round the numbers of `dist` (using `np.round()`), then calculate the mode.

In [5]:
rounded = np.round(dist)
mode = mode(rounded).mode[0]
print(mode)

7.0


C:\Users\shamu\AppData\Local\Temp\ipykernel_31064\2527650758.py:2: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode = mode(rounded).mode[0]


##### Q4: Determine if `dist` is skewed. If so, which way? How can you tell?

If the mean is greater than the median, the distribution is positively skewed. If the mean is less than the median, it's negatively skewed.

> Compare mean_dist and median_dist.

In [6]:
skewness = np.mean(dist) - np.median(dist)
if skewness > 0:
    skew_direction = "positively skewed"
elif skewness < 0:
    skew_direction = "negatively skewed"
else:
    skew_direction = "not skewed"
print(skew_direction, skewness)

positively skewed 8.04554455445545


##### Q5: Calculate the range of `dist`.

In [7]:
range = np.max(dist) - np.min(dist)
print(range)

167.0


##### Q6: Calculate the variance of `dist`.

In [8]:
variance = np.var(dist)
print(variance)

957.8005247034604


##### Q7: Calculate the standard deviation of `dist`. What does this tell you about the distribution of numbers? How does the standard deviation relate to the variance?

In [9]:
std_dev = np.std(dist)
print(std_dev)

30.94835253617647


## Relationship Between Standard Deviation and Variance

The standard deviation is the square root of the variance. They both measure the dispersion of the data points, but they are in different units. While variance gives a squared unit that may not be directly interpretable, the standard deviation is in the same unit as the data, making it easier to understand and compare with the mean.

**Range** - The range of 167.0 indicates the spread between the smallest and largest values in the dist dataset. A higher range often suggests greater variability in the data.

**Variance** - The variance of approximately 957.8 quantifies the dispersion of the data points around the mean. A higher variance indicates that the data points are more spread out from the mean.

**Standard Deviation** - The standard deviation of approximately 30.95 is the square root of the variance. It measures the average distance of each data point from the mean. A higher standard deviation indicates greater variability in the data.